Fix installer pre-check UI layout by using scrollable panel

divyesh-git22 opened the pull request:
This PR refactors the Pre-Install Checks UI to use a scrollable Panel
instead of fixed positioning.
what this change does:
- Prevents UI overflow on smaller screens
- Removes dependency on manual spacing
No installer logic was changed by myside

divyesh-git22 committed:
Fix installer pre-check UI layout by using scrollable panel

emtuls commented:
Thank you for the PR @divyesh-git22! 

I am testing the PR on Windows 10 and I did notice some issues that need to be addressed:

1. **The `Pre-Install Checks` window seems to be broken:**
Scroll wheels have been added to the sides, which is a great change! But unfortunately, some other things broke in the process:
- The checkbox we had to ensure users have made a snapshot before continuing seems to be missing.
- The `Continue` and `Cancel` boxes seem to be missing.
- The `True` and `False` indicators next to each check also seem to be missing.

Your code:
<img width="501" height="470" alt="image" src="https://github.com/user-attachments/assets/1ab95be7-8d1e-4dad-8972-9ca870b6cbc0" />
<img width="701" height="671" alt="image" src="https://github.com/user-attachments/assets/949e4b96-f3da-4ff5-98ae-8a75a6d35278" />

This is how I would expect it to look when fixed:
<img width="629" height="477" alt="image" src="https://github.com/user-attachments/assets/8a061bff-f01d-44a9-a7ab-9ec0771e4682" />

2. **The internet connectivity checks seem to have been removed:**

Before (original code):
<img width="580" height="134" alt="image" src="https://github.com/user-attachments/assets/1657bad4-3b39-4842-8512-b008347c1a3c" />

After (your PR):
<img width="586" height="214" alt="image" src="https://github.com/user-attachments/assets/2e1f8166-b2db-4e3e-9ec9-7433ba3f0129" />


3. **The package selection window still doesn't include the scroll bars on the sides, which I believe was the main concern from #737** 

Your code (note there are no scroll wheels on the sides for lower resolutions):
<img width="759" height="555" alt="image" src="https://github.com/user-attachments/assets/6a1444a7-1c3f-408f-8450-e6e6e9ebea12" />

This is how I would expect it to look when fixed:
<img width="912" height="863" alt="image" src="https://github.com/user-attachments/assets/5d3d101f-637c-41ac-a088-ee4d5761e6f6" />

emtuls reviewed (changes requested):
See previous comment!

PrajeetGuha commented:
@divyesh-git22 I checked your code. Looks like your edits are on pre-install check window and not on the installation wizard that has the package selection, installation path, etc. configurations.
My issue is with the **VM package selection** window that overflows and prevents me to click the OK button at the bottom as stated by @emtuls .

divyesh-git22 commented:
Hii @emtuls and @PrajeetGuha,

I sincerely apologize for the confusion and the broken UI in my previous commits. As I am new to the project, I misunderstood which window needed the scrollbar and accidentally broke the Pre-Install UI and internet check logic.

I am currently reverting install.ps1 to its original state to fix these errors. Once the project is back to a working state, I will correctly apply the scrollable fix to the Package Selection window as originally requested in #737.

Thank you for your patience and for the detailed feedback

emtuls commented:
Hi @divyesh-git22!

I see that you are making a bunch of commits which might be unnecessary. 🙂 Also, we prefer to only have 1 commit made per change (i.e., 1 commit for this PR) to keep a cleaner git history.

Might I suggest one of the following to clean up the branch?

**Option 1: Start fresh (Deletes your current changes).** 
Use this to wipe the slate and re-apply your changes cleanly:
- `git reset --hard main`
- make your changes to the file
- `git add <file>`
- `git commit -m "Clean, single commit message"`
- `git push origin <your-branch-name> --force-with-lease`


**Option 2: Squash and Reset (Keeps your work, but cleans history)**
Alternatively, this would revert you back to the main commit, keeping your changes, so that you can make a single commit:
- `git reset --soft main`
- `git commit -m "Clean, single commit message"`
- `git push origin <your-branch-name> --force-with-lease`

divyesh-git22 committed:
Fix #737: Added scrollbar and adjusted UI for Package Selection window

divyesh-git22 commented:
Hii @emtuls,

Thank you so much for your guidance and for helping me ! I sincerely apologize for the earlier confusion and any extra work it caused you, I appreciate your patience while I was learning the process.

I have now followed Option 2 and squashed all changes into a single, clean commit. I also moved the fix to the correct Package Selection window as requested:

Added a scrollbar to the categories panel.

Adjusted the window size and moved the buttons (Install/Cancel) up to ensure they are visible on all screen resolutions.
The PR should be much cleaner and ready for your review now. Thank you again for all your help
